defaults/cmake/Key4hepConfig.cmake

```
@@ -11,7 +12,10 @@ macro(key4hep_set_compiler_flags)
     set(COMPILER_FLAGS "${COMPILER_FLAGS} -fdiagnostics-color=always")
```

```suggestion
    set(COMPILER_FLAGS "${COMPILER_FLAGS} -fdiagnostics-color=auto")
```
As far as I can remember `always` breaks emacs compilation-mode parsing.
For the CI we anyway add Werror, and could add `always` as well.

---

This I did on purpose because `auto` does not work when I'm connected through ssh and run things in containers while `always` does. On emacs it works fine for me with `always`, I get colored output. If it doesn't break anything I would prefer to have `always` by default.

---

Ok, indeed seems emacs learned to parse the color output :)